### PR TITLE
Lazy-load all relations

### PR DESCRIPTION
**Pull request by davidmehren**
### Component/Part
entities

### Description
This PR refactors the entity classes to lazy-load all relations.

### Steps

- [x] Added implementation
- [x] Added / updated tests
- [x] I read the [contribution documentation](https://github.com/hedgedoc/hedgedoc/blob/develop/CONTRIBUTING.md) and
  made sure that:
  - My commits are signed-off to accept the DCO.
  - This PR targets the correct branch: `master` for 1.x & docs, `develop` for 2.x

### Related Issue(s)
Closes https://github.com/hedgedoc/hedgedoc/issues/1642

### CODE REVIEW

**Review by InnayTool — APPROVED** [review]

File: src/auth/auth.service.ts [hunk 1]
```diff
@@ -53,17 +53,21 @@ export class AuthService {
...
+    );
   }
 
   async createTokenForUser(
     user: User,
     identifier: string,
     validUntil: TimestampMillis,
   ): Promise<AuthTokenWithSecretDto> {
-    user.authTokens = await this.getTokensByUser(user);
+    user.authTokens = this.getTokensByUser(user);
```
Comment (DerMolly) [inline src/auth/auth.service.ts:68]: Maybe a type hint would be useful here?

File: src/notes/alias.service.ts [hunk 2]
```diff
@@ -87,7 +87,7 @@ export class AliasService {
     let oldPrimaryId = '';
     let newPrimaryId = '';
 
-    for (const anAlias of note.aliases) {
+    for (const anAlias of await note.aliases) {
```
Comment (DerMolly) [inline src/notes/alias.service.ts:90]: Maybe it would be wise to await this once and then put it in a variable.

**Review by DerMolly — CHANGES_REQUESTED** [review]

File: src/auth/auth.service.ts (hunk 1 above)
Comment (davidmehren) [reply src/auth/auth.service.ts:68]: I'm not declaring a variable here (the `User` class defines the type of the property), so I can't an explicit type.

File: src/notes/alias.service.ts (hunk 2 above)
Comment (davidmehren) [reply src/notes/alias.service.ts:90]: I'm pretty sure that this awaits `note.aliases` only once :smile:

File: src/media/media.service.spec.ts [hunk 3]
```diff
@@ -222,9 +222,9 @@ describe('MediaService', () => {
         const mockMediaUploadEntry = {
           id: 'testMediaUpload',
           backendData: 'testBackendData',
-          user: {
+          user: Promise.resolve({
             username: 'hardcoded',
```
[the remark this answers is not in the document]
Comment (davidmehren) [reply src/media/media.service.spec.ts]: done

**Review by DerMolly — APPROVED** [review]

**Pull request merged**